### DISCUSSION TITLE
Increasing ShineWIFI-X (S) report frequency

### DISCUSSION THREAD

**Discussion by tabascoz** [johanmeijer/grott#93] [Show and tell]
Hi @johanmeijer and everyone who might be interested

I was able to decrease ShineWIFI-X frequency from 5min to 1min by trial and error, changing parameter "4" from 5 to 1. 

If anyone is willing to try ( at your own risk ), go to Growatt website > Dashboard > My Photovoltaic Devices "all devices" (top right link) > data logger settings. 

Then enter the key, advanced set and input register 4 with value 1. 
![Screenshot from 2022-02-01 22-41-02](https://user-images.githubusercontent.com/10350083/152080301-ed90f1f0-6c23-4975-9e33-1b37125af137.png)

**Comment by mjdyson** [comment]
Good find! I've also noticed that this setting might be available if you log in directly to the data logger (admin/admin). I had to find the IP address via the DHCP table in my router. I wonder if this is the same parameter?

![image](https://user-images.githubusercontent.com/4598068/152129908-750d40e3-3597-45b4-9caa-39d27847c893.png)

**Comment by johanmeijer** [comment]
Yes this working (the register setting via de growatt server site). Be aware you have to do it in the data logger setting menu and not in the inverter settings menu. There are two type of registers: data logger and inverter. The data logger settings are not documented vary well. 

The inverter registers are available (and probably in this way you can enable and disable some things like battery charging etc). 

I was trying to find out which register is what and want to document this (and implement it in Grott as requested in issue #73). 

Be aware if you enabled command blocking in grott (blockcmd=True) it will not work.

**Comment by mjdyson** [comment]
Has anyone experimented with lower frequencies in reporting? ie, a value of 0.05 would give data every 3 seconds - perhaps a bit better for driving a real time display of local MQTT reporting. I have something similar with local CT clamps into emoncms, but I cant get the battery data that way.

**Comment by johanmeijer** [comment]
If I set Register 4 to a value below 1 (e.g 0.5, 0.1) I see that the datalogger sent the same record multiple times (no fresh measurements). 
So I think 1 is really the minimum (at least for my configuration).

**Reply by tabascoz** [reply]
Will try it on spf3500es this weekend and report back

**Comment by cspiby** [comment]
Is it not sufficient to just change it in the web server of the Shinelan? I have gone on to the website and attempted to read that register without first setting it (to see what its current value is) and I just get an error.

![Screenshot 2022-02-15 203444](https://user-images.githubusercontent.com/29413212/154144741-af621fcf-50e4-47db-98a3-eec3679f0d2e.png)

**Reply by johanmeijer** [reply]
If possible in the shinelan web server then that might be better. My shinewifi does not have a web server interface onze configure (have to reset it again).

Wrong key is because you have to specify the key (growatt +date, eg growatt20220215).

**Reply by cspiby** [reply]
Changing update frequency in the shinelan to 1 minute doesnt seem to have an effect, even though it even shows in the growatt dashboard on the website, strange.

**Comment by johanmeijer** [comment]
I did some tests with grottserver.py (see discussion https://github.com/johanmeijer/grott/discussions/98) as destination in stead of a connection to the growatt server. My conclusion : The growatt server is the limitation when specifying a report frequency <= 1 minute. Lower then that may cause connection errors. 

With the beta grotserver.py running at the same machine as grott 1 minute is no problem. I even manged to run with 0.5 (half a minute) and 0.1 (every 6 seconds) without any problems.

Be aware this will increase the PVOutput request limit problem (only 60 request per hour are allowed)  A reason more to look for a solution for that ;)

**Reply by jupppo** [reply]
Hi there. Could be grott configured in a way that it sends out data to growatt on a 5/1min bases (ideally avg), but makes local polls available in a much shorter time?

**Comment by johanmeijer** [comment]
At this moment that is not possible. Grott does not poll but "reads" the data being sent to grotwatt (or the grottserver). 
You can increase the freqency but setting the register as defined in this thread (or via an API in grottserver). 

Growatt does not like high frequent updates so increasing your interval lower then 2 minutes will be a challenge. With grottserver the interval can  be much  shorter but the you loose the possibility to write to growatt (at this moment, will be probably available in the future).
